### PR TITLE
add vitest, add json validation tests, fix broken json files

### PR DESCRIPTION
**Pull request by havardthom**
> [!NOTE]
> We are meticulous when it comes to merging code into the main branch, so please understand that we may reject pull requests that do not meet the project's standards. It's never personal. Also, game-related scripts have a lower chance of being merged.

## Description

- Add [vitest](https://vitest.dev/) and [React Testing Library](https://testing-library.com/) to frontend
- Implement json validation tests
  - test valid json syntax
  - test valid schema
  - test that script file exist for each install method in json
- Fix broken json files
  - Change `interface_port`, `hdd`, `cpu` and `ram` from string to number
  - Some other small bugs
- Change OS version in schema from number to string

Will add linting and testing to Github action later which will be used to validate PR's

## Type of change
Please check the relevant option(s):

- [x] Bug fix (non-breaking change that resolves an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (a fix or feature that would cause existing functionality to change unexpectedly)
- [ ] New script (a fully functional and thoroughly tested script or set of scripts.)

## Prerequisites
The following efforts must be made for the PR to be considered. Please check when completed:
- [x] Self-review performed (I have reviewed my code, ensuring it follows established patterns and conventions)
- [x] Testing performed (I have tested my changes, ensuring everything works as expected)
- [x] Documentation updated (I have updated any relevant documentation)

## Additional Information (optional)
Provide any additional context or screenshots about the feature or fix here.

green

example broken json:
invalid-json

example broken schema:
invalid-schema

## Related Pull Requests / Discussions

If there are other pull requests or discussions related to this change, please link them here:
- Related PR #

### CODE REVIEW

File: frontend/src/app/json-editor/page.tsx
Comment (havardthom) [inline frontend/src/app/json-editor/page.tsx:1]: only change in this file is `type Script`. The rest is linting

**Comment by havardthom** [comment]
Json files removed because they don't have scripts:
```
collabora-online.json
collabora.json
scrutiny.json
```

**Comment by havardthom** [comment]
Fixed json files with bug in `install_method.script` path
kimai.json 
nextpvr.json

**Review by MickLesk — APPROVED** [review]

**Comment by MickLesk** [comment]
json editor is also customized? So that it is no longer hdd=“2”, instead hdd=2 ?

**Comment by havardthom** [comment]
> json editor is also customized? So that it is no longer hdd=“2”, instead hdd=2 ?

json editor already used `number` for these fields, only the json files was not following the schema

**Review by bramsuurdje — APPROVED** [review]

**Comment by BramSuurdje** [comment]
LGTM! good addition

**Pull request merged**